### PR TITLE
update(grid.py): add size property for model splitting support

### PR DESCRIPTION
**Pull request by jlarsen-usgs**

### CODE REVIEW

**Comment by mwtoews** [comment]
A [simple one-liner](https://stackoverflow.com/a/54697819/) could also work:
```python
import math
...
    @property
    def size(self):
        return math.prod(self.shape)
```
a test on both a structured and unstructured grid should be added to this PR, e.g. in `autotest/test_grid.py`, add `assert mg.shape == <exepcted>` and `assert mg.size == <expected>` for `test_structured_xyz_intersect` and `test_unstructured_xyz_intersect`.

**Comment by jlarsen-usgs** [comment]
@mwtoews 
Thanks for the suggestions. numpy.prod() would be much simplier!

File: autotest/test_grid.py [hunk 1]
```diff
@@ -259,6 +259,9 @@ def test_structured_xyz_intersect(example_data_path):
         model_ws=str(example_data_path / "freyberg_multilayer_transient"),
     )
     mg = ml.modelgrid
+
+    assert mg.size == np.prod((mg.nlay, mg.nrow, mg.ncol))
```
Comment (mwtoews) [inline autotest/test_grid.py:263]:
```suggestion
    assert mg.shape == (3, 40, 20)
    assert mg.size == 2400
```

File: autotest/test_grid.py [hunk 2]
```diff
@@ -281,6 +284,8 @@ def test_vertex_xyz_intersect(example_data_path):
     ml = sim.get_model(list(sim.model_names)[0])
     mg = ml.modelgrid
 
+    assert mg.size == np.prod((mg.nlay, mg.ncpl))
```
Comment (mwtoews) [inline autotest/test_grid.py:287]:
```suggestion
    assert mg.shape == (4, 100)
    assert mg.size == 400
```

File: autotest/test_grid.py [hunk 3]
```diff
@@ -334,6 +339,8 @@ def test_unstructured_xyz_intersect(example_data_path):
         ncpl=ncpl,
     )
 
+    assert mg.size == mg.nnodes
```
Comment (mwtoews) [inline autotest/test_grid.py:342]:
```suggestion
    assert mg.shape == (654,)
    assert mg.size == 654
```

**Review by mwtoews — APPROVED** [review]
Looks good, just a few suggestions to improve the tests, which should test the expected value. It's also handy to test the expected shape.

**Review by langevin-usgs — APPROVED** [review]

**Pull request merged**